src/jdk.incubator.vector/share/classes/jdk/incubator/vector/ByteVector.java

```diff
@@ -2279,14 +2279,21 @@ public final ByteVector blend(long e,
     /*package-private*/
     final
     @ForceInline
-    ByteVector sliceTemplate(int origin, Vector<Byte> v1) {
-        ByteVector that = (ByteVector) v1;
-        that.check(this);
+    ByteVector sliceTemplate(int origin, Vector<Byte> v) {
         Objects.checkIndex(origin, length() + 1);
-        VectorShuffle<Byte> iota = iotaShuffle();
-        VectorMask<Byte> blendMask = iota.toVector().compare(VectorOperators.LT, (broadcast((byte)(length() - origin))));
-        iota = iotaShuffle(origin, 1, true);
-        return that.rearrange(iota).blend(this.rearrange(iota), blendMask);
+        ByteVector that = (ByteVector) v;
+        that.check(this);
+        return VectorSupport.slice(
+            getClass(), byte.class, length(),
+            this, that, origin,
+            new VectorSliceOp<ByteVector>() {
```

Comment:
Change from inner class to lambda expression?

Reply:
We still need this method to be inlined and I don't know if there is a way to annotate the lambda function.

Reply:
Yes, i wondered about the inline and how important it might be. You want the fallback to inline so as not to perturb platforms without the intrinsic. Can you add a comment on the anon class?